src/ConditionalTokensMapping.ts

```diff
@@ -192,6 +200,49 @@ export function handleConditionResolution(event: ConditionResolution): void {
   condition.payoutNumerators = payoutNumerators;
   condition.payoutDenominator = payoutDenominator;
   condition.resolutionHash = event.transaction.hash;
-  
+  let marketMakers = condition.fixedProductMarketMakers;
+  for (let i = 0; i < marketMakers.length; i += 1) {
+    // This is not ideal as in theory we could have multiple market makers for the same condition
+    // Given that this subgraph only tracks market makers deployed by Polymarket, this is acceptable for now
+    let fpmm = FixedProductMarketMaker.load(
+      marketMakers[i],
+    ) as FixedProductMarketMaker;
+    let collateralScale = getCollateralScale(fpmm.collateralToken);
+    let collateralScaleDec = collateralScale.toBigDecimal();
+
+    if (fpmm) {
+      let marketPositions = fpmm.marketPositions;
+      if (marketPositions as Array<string>) {
+        for (let j = 0; j < marketPositions!.length; j += 1) {
+          let pos = marketPositions![j];
+          let position = MarketPosition.load(pos);
+          if (position) {
+            let numerator = payoutNumerators[position.outcomeIndex.toI32()];
+            let redemptionValue = position.netQuantity
+              .times(numerator)
+              .div(payoutDenominator);
+            let averageRedemptionPrice = redemptionValue.div(
+              position.netQuantity,
+            );
+            let averagePricePaid = position.netValue.div(position.netQuantity);
```

Comment:
netValue is set when you buy?

Reply:
netValue is the total amount the user has paid to get into the current position, unique per account-market-outcome

It gets increased on buys and decreased on sells.

Reply:
So you continue to impact netValue and netQuantity on subsequent buys, sells, merges, splits